### DISCUSSION TITLE
Python NoneType object of __builtin__module

### DISCUSSION THREAD

**Discussion by larscordes** [community/community#21379] [Other Feature Feedback, Questions, & Ideas]
<p>Hi there,</p>
<p>I’m trying to do something extremely simple. I have a list and want to append a variable to it (in the middle of a for loop).</p>
<p>i = “c”<br>
matchlist = [“a”,“b”]<br>
i = str(i)<br>
matchlist = matchlist.append(i)</p>
<p>I expected to get a list ‘matchlist’ with two records (‘a’ and ‘b’) in it and then append a new record ‘c’ to that list. However instead the list loses its type and the value comes up as ‘NoneType object of __builtin__ module’ (I’m using Spyder).</p>
<p>Any pointers on where I’m going wrong?</p>
<p>Many thanks.</p>

**Comment by subramaniam02** [comment, marked as answer]
<p>you dont have to do matchlist = matchlist.append(i)</p>
<p>matchlist is an array, so just append the element</p>
<p>reassigning matchlist = matchlist.append(i) will make matchlist as None because append method returns None</p>

**Comment by larscordes** [comment]
<p>Great thanks - very helpful!</p>
